packages/forms/src/UIForm/fields/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,190 @@
+import PropTypes from 'prop-types';
+import memoize from 'lodash/memoize';
+import InputDateTimePickerComponent from '@talend/react-components/lib/DateTimePickers';
+import FieldTemplate from '../FieldTemplate';
+import { isoDateTimeRegExp } from '../../customFormats';
+import { UnhandleTypeError, UnexpectedTypeError } from './WrongTypeError';
+
+export const GENERIC_FORMAT_ERROR = 'GENERIC FORMAT ERROR';
+
+const INVALID_DATE = new Date('');
+const HANDLE_CONVERTION_TYPE = ['string', 'number'];
+
+function isDateValid(date) {
+	if (date === undefined) {
+		return true;
+	}
+
+	return date instanceof Date && !isNaN(date.getTime());
+}
+
+function convertDateToTimestamp(date) {
+	return date.getTime();
+}
+
+function convertTimestampToDate(timestamp) {
+	return new Date(timestamp);
+}
+
+function convertDateToString(date) {
+	return date.toISOString();
+}
+
+function convertStringToDate(str) {
+	if (!isoDateTimeRegExp.test(str)) {
+		return INVALID_DATE;
+	}
+
+	return new Date(str);
+}
+
+function convertToDate(type, value) {
+	if (value === undefined) {
+		return undefined;
+	}
+
+	const typeOfValue = typeof value;
+
+	if (typeOfValue !== type) {
+		// eslint-disable-next-line no-console
+		console.error(new UnexpectedTypeError(type, typeOfValue));
+		return INVALID_DATE;
+	}
+
+	switch (type) {
+		case 'number':
+			return convertTimestampToDate(value);
+		case 'string':
+			return convertStringToDate(value);
+		default:
+			// eslint-disable-next-line no-console
+			console.error(new UnhandleTypeError(HANDLE_CONVERTION_TYPE, type));
+			return INVALID_DATE;
+	}
+}
+
+function convertFromDate(type, date) {
+	if (date === undefined) {
+		return undefined;
+	}
+
+	switch (type) {
+		case 'number':
+			return convertDateToTimestamp(date);
+		case 'string':
+			return convertDateToString(date);
+		default: {
+			const unhandleTypeError = new UnhandleTypeError(HANDLE_CONVERTION_TYPE, type);
+			// eslint-disable-next-line no-console
+			console.error(unhandleTypeError);
+			return unhandleTypeError;
+		}
+	}
+}
+
+class InputDateTimePicker extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			value: props.value,
+		};
+
+		this.onChange = this.onChange.bind(this);
+		this.onBlur = this.onBlur.bind(this);
+		this.convertToDate = memoize(convertToDate, (type, value) => `${type}||${value}`);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		const newValue = nextProps.value;
+		if (newValue !== this.state.value && !(newValue instanceof Error)) {
+			this.setState({
+				value: newValue,
+			});
+		}
+	}
+
+	/**
+	 * On change callback
+	 * @param date
+	 */
+	onChange(event, errorMessage, date) {
+		const { schema } = this.props;
+		const type = schema.schema.type;
+
+		const hasError = errorMessage !== undefined;
+
+		const value = hasError ? new Error(errorMessage) : convertFromDate(type, date);
+
+		const payload = {
+			schema: this.props.schema,
+			value,
+		};
+		this.props.onChange(event, payload);
```

Comment:
I don't know what your really asking ? There is 0 return path ?!
If your talking about the multiple possible values given to the function, and you want to have as many call declaration as possible values for the same function, please explain why :). I just don't understand why it could be a good thing ?!

Reply:
❓